### ISSUE TITLE
Possible Information Leakage in Prompt Reshaping During Testing

### ISSUE THREAD

**Issue by xvrong** [zhoujiahuan1991/AAAI2025-CAPrompt#2]
您好，你们的工作展示了一个令人惊艳的结果，对我有很大的启发。但是我似乎发现了代码中的一处问题

https://github.com/zhoujiahuan1991/AAAI2025-CAPrompt/blob/24f854b4b48e82a33ca249e736e8ea23a9f22f1c/prompt/caprompt.py#L71-L75

这里batched_prompt_raw在reshape前后的维度含义不能一一对应，我认为这会导致测试时样本间信息的泄漏，导致同一个batch的样本在测试时近似于投票选出他们所要使用的prompt。在训练时这可能不会引起什么问题，但是由于测试是逐个任务进行的，样本间信息的泄漏将会对结果造成很大的影响，而且论文里使用的循环迭代的测试方法会很大程度上加剧这种泄漏。

我对此进行了一些测试。首先，我仅将测试时的batch size设为1，观察到了极大的性能损失。进一步的，如果在reshape之前执行transpose将batched_prompt_raw的维度一一对齐，重新进行训练，不需要更改测试时的batch size就可以观察到与上文所述相近的性能，在ImageNet-R数据集上的ACC大概为72左右。

我认为这一问题似乎违反了增量学习上大家通用的实验设定，请问是我搞错了什么吗？

**Comment by marua172** [comment]
Hi,

I have the same question as @xvrong .

The batch-wise testing accuracy is generally much higher than the instance-wise accuracy. This point is also discussed on page 7 of the following paper:
[https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&amp;arnumber=10820551](https://ieeexplore.ieee.org/stamp/stamp.jsp%3Ftp%3D%26arnumber%3D10820551)

Could the authors please clarify if the results for instance-wise testing are available or if such tests were conducted?

Thank you!

**Comment by qiweili00** [comment]
Apologies for the delayed response.

Thank you for your interest in our work. We directly adopt the same training and testing setup as used in [DualPrompt](https://github.com/JH-LEE-KR/dualprompt-pytorch/blob/7eb457d988409a6abf97af2b121ffa62dd4b498a/prompt.py#L118) and [HiDe-Prompt](https://github.com/thu-ml/HiDe-Prompt/blob/0ed433334ff2cfb206c5e2899651c3044ec85f92/peft/prompt/hide_prompt.py#L100).


We noticed that similar questions have also been discussed in this [issue](https://github.com/aimagelab/mammoth/issues/56). In addition to the batch-size testing, we plan to check the code and verify the results for instance-wise testing. Due to our current busy schedule, we expect to provide an update around June.
